geomesa-convert/geomesa-convert-json/src/test/scala/org/locationtech/geomesa/convert/json/JsonConverterTest.scala

```diff
@@ -1542,5 +1544,105 @@ class JsonConverterTest extends Specification {
         }
       }
     }
+
+    "infer schema from non-geojson files" >> {
+      val json =
+        """{ "name": "name1", "demographics": { "age": 1 }, "files":[1], "geom": "POINT (41 51)" }
+          |{ "name": "name2", "files":[2,3], "geom": "POINT (42 52)" }
+          |{ "name": "name3", "demographics": { "age": 3 }, "files":[4,5,6], "geom": "POINT (43 53)" }
+        """.stripMargin
+
+      def bytes = new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8))
+
+      val factory = new JsonConverterFactory()
+      // test deprecated infer method as well as new one
+      val old = factory.infer(bytes).fold[Try[(SimpleFeatureType, Config)]](Failure(null))(Success(_))
+      foreach(Seq(factory.infer(bytes, None, Map.empty[String, AnyRef]), old)) { inferred =>
```

Comment:
I think it would be better if we could split this into two separate unit tests - one for the deprecated infer method and one for the new method - if possible.